app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,93 @@
+require 'pry'
+class TweetsController < ApplicationController
+    
+   
+    get '/tweets' do
+        if !logged_in?
+          redirect '/login'
+      else 
+        @tweets = Tweet.all 
+        erb :'tweets/index'
+        end
+    end 
+    
+    
+    get '/tweets/new' do
+        if logged_in?
+        erb :'tweets/new' #show new tweets view
+    else 
+        redirect '/login'
+        end
+    end 
+    
+    
+    post '/tweets' do
+     if logged_in?
+        if params[:tweet][:content] != ""
+            @tweet = Tweet.new(params[:tweet]) #create new tweet
+            @tweet.user_id  = current_user.id #set session to the users id
+            
+            if @tweet.save #saves new tweet 
+                redirect '/tweets' #redirect back to tweets index page
+            else
+                erb :'tweets/new' 
+            end
+        else 
+            redirect '/tweets/new'
+        end
+    else
+        redirect '/login'
+     end
+    end
+
+
+    
+    get '/tweets/:id' do
+      if !logged_in?
+        redirect '/login'
+      else 
+       @tweet = Tweet.find(params[:id]) 
+       erb :'tweets/show_tweet' #show single tweet view
+      end
+    end 
+    
+    
+    
+    get '/tweets/:id/edit' do
+      if !logged_in?
+        redirect '/login'
+      else 
+        @tweet = Tweet.find(params[:id]) 
+        erb :'tweets/edit_tweet' #show edit tweet view
+      end
+    end 
+    
+    
+    patch '/tweets/:id' do
+       @tweet = Tweet.find(params[:id])
+    if params[:tweet][:content] != ""
+       @tweet.assign_attributes(params[:tweet]) #assign tweet attributes
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...